File: 프로그래머스/lv1/1845. 폰켓몬/폰켓몬.js
function solution(nums) {
    let set = new Set();
    
    for(let i=0; i<nums.length; i++){
        set.add(nums[i])
    }
    if(set.size < nums.length/2){
        return set.size;
    }else{
        return nums.length/2
    }
}

File: 프로그래머스/1/1845. 폰켓몬/폰켓몬.java
import java.util.*;

class Solution {
    public int solution(int[] nums) {
        int choice = nums.length/2;
        
        HashMap<Integer, Integer> map = new HashMap<>();
        
        for(int i=0; i<nums.length; i++){
            map.put(nums[i],0);
        }
        int kind = map.keySet().size();
        
        if(kind <= choice){
            return kind;
        }
        return choice;
    }
}